Predicates update with language and resource type implementation. #2 #14

lsitu opened the pull request:
https://github.com/ucsdlib/dams5-cc-pilot/issues/2
https://github.com/ucsdlib/dams5-cc-pilot/issues/14

lsitu committed:
Added language with URL validation and assiociate it with th forms/UI
for Object and Collection. #2

lsitu committed:
Updated predicates for rightsOverride, type etc and implemented resource
type uri with CVs and labels. #14

lsitu commented:
@ucsdlib/developers Please review and comments. Thanks.

mcritchlow commented on app/services/metadata_service.rb:
```diff
@@ -39,8 +39,22 @@ def local_attribution_list
 
-    def format_list
-      ['Cartographic', 'Data', 'Image', 'Notated music', 'Text', 'Video', 'Sounding recordings-nonmusical']
+    def resource_type_list
```
I wonder if we could, eventually, turn this into a rake task or something that would use RDF.rb functionality to run through a list of URI's, fetch the graphs, and extract the labels. Not for this PR of course, but when we move to building out the new application.

Similar to the [gist](https://gist.github.com/mcritchlow/e36e841b9e6a019a971c6ce7858cb555/edit) I mocked up earlier when we were discussing subjects, but grabbing the labels instead 

While I doubt this list would change often, in other contexts the lists might. So not having them hard-coded would be interesting. Or, even allowing curators to manage the lists via the UI. It would also remove the possibility of mis-spelled labels that would require a deployment to fix, such as Mixed material missing the final 'l' character in this list.

lsitu replied:
@mcritchlow : Yes, I agree. I think one way that we can do is to add some special class like "ResourceType" to extend skos:Concept or the basic Authority class (with label and URI only), then we can simply management them through the local authority management system.

lsitu replied:
And good catch for the typo "materia"! Thanks. I just update the PR to fix it.

lsitu committed:
Updated to create local authority to manage ResourceType properties that
use xsd:uri with control value list. #2 #14

lsitu commented:
@mcritchlow : I've updated the PR to add [commit](https://github.com/ucsdlib/dams5-cc-pilot/pull/25/commits/f1ac5b19757462f5080ff1a2877674017651a2ee), which enable us to create local authority records with form/UI to manage those properties like Resource Type that are URI with control value list. And I also added support to import all the ResourceType CVs by running the authority import tool `bin/import-authorities`

mcritchlow reviewed (approved):
👍 LGTM.

I like the way the CV's reside in the import-authorities script. That would allow us to potentially sub out the hard-coded lists with more dynamic use of RDF/linked data dereferencing in the future.

VivianChu commented:
👍

hweng commented:
:+1:

lsitu commented:
Thank you all @ucsdlib/developers for your reviews and comments.
